pkg/services/publicdashboards/service/service.go

```diff
@@ -342,6 +342,32 @@ func (pd *PublicDashboardServiceImpl) Delete(ctx context.Context, orgId int64, u
 	return nil
 }
 
+func (pd *PublicDashboardServiceImpl) DeleteByDashboard(ctx context.Context, dashboard *dashboards.Dashboard) error {
+	if dashboard.IsFolder {
```

Comment:
I think all of this should be in the same transaction. Thoughts? @evictorero

Reply:
What would the downside be if we only deleted some of the public dashboards and then returned an error?

Reply:
have done an inconsistent deletion. From my point of view, everything should be deleted or if there's an error in any deletion, anything should be deleted and an error should be thrown. Otherwise, we end up in an inconsistent state which we don't know how to manage. It's my thought. We should decide what to do but why not to make it as a transaction if we will throw the error anyways?

Reply:
If we are not blocking the dashboard deletion (which makes sense) I would say it's the same, maybe it is even better to delete a few ones since they will be orphans.
I'm wondering what we should do if the dashboard deletion fails after that.

Reply:
I think we should only use transactions when we want to avoid a weird state. For example, if we start deleting many pubdashes for a dashboard folder then fail halfway through, we'll return an error and there will still be some pubdashes remaining. I don't think this leaves us in a bad state - we just have less pubdashes than when we started, and we're still returning an error letting the caller know something went wrong. Win-win.

Reply:
I think this boils down to our definition of what getting left in a `weird state` is.

Reply:
I was thinking of calling the public dashboard deletion after dashboard deletion success. Since we don't have FKs we can delete the father before the child. If there is no dashboard but we have public dashboard orphans it's ok, the other way we could delete a public dashboard but if the dashboard deletion fails some configuration will disappear (pubdash) and that is not consistent.

Reply:
currently we cant delete the pubdash after the dashboard bc we need to query the dashboard table when its a folder that got deleted